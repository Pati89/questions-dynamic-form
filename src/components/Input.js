import PropTypes from "prop-types";
import styles from "../styles/components/Input.module.css";

const Input = (props) => {
  const { field, onChange, value } = props;

  const { name, label } = field;

  // We can add here a validation for correct field types and if required.
  // At the moment all field types are text type

  return (
    <div className={styles.inputField} data-testid="input">
      <label htmlFor={name}>{label}</label>
      <input
        {...props}
        type="text"
        id={`${name}-input`}
        name={name}
        onChange={(event) => onChange(event, field.name)}
        value={value || ""}
        data-testid="input-field"
      />
    </div>
  );
};

Input.propTypes = {
  field: PropTypes.object.isRequired,
};

export default Input;
