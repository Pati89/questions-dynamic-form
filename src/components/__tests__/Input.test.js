import Input from "../Input";
import { render, screen } from "@testing-library/react";

const props = {
  field: { name: "test_field", label: "Test Field", type: "text" },
};

describe("Input", () => {
  test("should render a Input", () => {
    render(<Input {...props} />);
    const input = screen.getByTestId("input");
    expect(input).toBeTruthy();
    expect(input).toHaveTextContent("Test Field");
  });

  test("should pass name prop", () => {
    render(<Input {...props} />);
    const inputField = screen.getByTestId("input-field");
    expect(inputField).toBeTruthy();
    expect(inputField).toHaveProperty("name", "test_field");
  });

  test("should render a Input match snapshot", () => {
    render(<Input {...props} />);
    const input = screen.getByTestId("input");
    expect(input).toMatchSnapshot();
  });
});
