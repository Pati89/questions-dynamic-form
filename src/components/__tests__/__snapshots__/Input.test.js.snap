// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render a Input match snapshot 1`] = `
<div
  class="inputField"
  data-testid="input"
>
  <label
    for="test_field"
  >
    Test Field
  </label>
  <input
    data-testid="input-field"
    field="[object Object]"
    id="test_field-input"
    name="test_field"
    type="text"
    value=""
  />
</div>
`;
